sda-commons-server-s3-testing/src/main/java/org/sdase/commons/server/s3/testing/S3InMemoryMock.java

```diff
@@ -0,0 +1,121 @@
+package org.sdase.commons.server.s3.testing;
+
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.List;
+import org.junit.rules.ExternalResource;
+import org.sdase.commons.server.s3.testing.builder.MockObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class S3InMemoryMock extends ExternalResource {
```

Comment:
Should this class be abstract?
If it is not too hard I'd try to remove the JUnit 4 "ExternalResource" superclass.